### PR TITLE
common: Fix compilation warnings replace deprecated functions gtk_rc_get_theme_dir and gtk_rc_scanner_new

### PR DESCRIPTION
**Pull request by zhuyaliang**

### CODE REVIEW

**Review by lukefromdc — APPROVED** [review]
Builds and runs fine, themes are properly rendered in mate-appearance-properties, which I am presuming to be the main user of the theme tester. Current GTK theme properly applied to the control center itself.

We get these four runtime warnings from mate-appearance-properties (with the newer glib versions only) but they are shared with master, need to be fixed elsewhere in the code, and can be a separate PR. Opened issue https://github.com/mate-desktop/mate-control-center/issues/735 for it:

(mate-appearance-properties:34711): GLib-GIO-CRITICAL **: 15:12:11.817: GFileInfo created without standard::type

(mate-appearance-properties:34711): GLib-GIO-CRITICAL **: 15:12:11.817: file ../gio/gfileinfo.c: line 1611 (g_file_info_get_file_type): should not be reached

(mate-appearance-properties:34711): GLib-GIO-CRITICAL **: 15:12:11.817: GFileInfo created without standard::type

(mate-appearance-properties:34711): GLib-GIO-CRITICAL **: 15:12:11.817: file ../gio/gfileinfo.c: line 1611 (g_file_info_get_file_type): should not be reached
No such key “theme”

File: capplets/common/gtkrc-utils.c [hunk 1]
```diff
@@ -176,6 +176,48 @@ void gtkrc_get_details(gchar* filename, GSList** engines, GSList** symbolic_colo
...
+  TRUE          /* scan_hex_dollar */,
+  TRUE          /* scan_string_sq */,
+  TRUE          /* scan_string_dq */,
+  TRUE          /* numbers_2_int */,
+  FALSE         /* int_2_float */,
+  FALSE         /* identifier_2_string */,
+  TRUE          /* char_2_token */,
+  TRUE          /* symbol_2_token */,
+  FALSE         /* scope_0_fallback */,
+};
```
Comment (raveit65) [inline capplets/common/gtkrc-utils.c:219]: New build warning.
debian
https://app.travis-ci.com/github/mate-desktop/mate-control-center/jobs/612020554#L4528
fedora
https://app.travis-ci.com/github/mate-desktop/mate-control-center/jobs/612020555#L1869
```
gtkrc-utils.c:219:1: warning: missing field 'store_int64' initializer [-Wmissing-field-initializers]

};

^

1 warning generated.
```
This is a normal build warning. I do not get why it is approved with a build warning which everybody can see on normal build.
And why nobody takes a lot at travis-ci build logs.

Comment (raveit65) [reply capplets/common/gtkrc-utils.c:219]: Ok sorry, build warning isn't visible with normal warn cflags. 
But in travis-ci.

**Comment by raveit65** [comment]
gtkrc is for gtk-2.0 themes. See https://github.com/mate-desktop/mate-themes/tree/master/desktop-themes/BlueMenta/gtk-2.0 for example. So do we need this code with gtk-3.0 themes?

**Comment by zhuyaliang** [comment]
> gtkrc is for gtk-2.0 themes. See https://github.com/mate-desktop/mate-themes/tree/master/desktop-themes/BlueMenta/gtk-2.0 for example. So do we need this code with gtk-3.0 themes?

I know little about the theme, and I don't know how to add gtk3 support

File: capplets/common/gtkrc-utils.c (hunk 1 above)
Comment (zhuyaliang) [reply capplets/common/gtkrc-utils.c]: done

**Comment by raveit65** [comment]
Opps, i did a force push by accidental but i was able to restore previous state.
I hope code is now OK.

**Review by raveit65 — APPROVED** [review]
LGTM,
m-a-p runs like before and deprecated warnings are gone.

**Pull request merged**